### PR TITLE
Add 'ExpandValueTuple' as a misc symbol display flag

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/66757

### CODE REVIEW

File: src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayCompilerInternalOptions.cs [hunk 1]
```diff
@@ -53,31 +53,26 @@ internal enum SymbolDisplayCompilerInternalOptions
...
-        /// <summary>
-        /// Display `System.ValueTuple` instead of tuple syntax `(...)`.
-        /// </summary>
-        UseValueTuple = 1 << 6,
-
         /// <summary>
         /// Display `System.[U]IntPtr` instead of `n[u]int`.
         /// </summary>
-        UseNativeIntegerUnderlyingType = 1 << 7,
+        UseNativeIntegerUnderlyingType = 1 << 6,
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayCompilerInternalOptions.cs:59]: i changed these values to keep things contiguous (they're internal, so no breaking change here).  can keep them the same though if that is preferred.

**Comment by CyrusNajmabadi** [comment]
Ick. a bunch of tests change here because they were passing in a common error format with the "ExpandValueTuple" flag which VB previously didn't support.  Now that it does, the tests all change in their error output (in generated IL comments).  What's the preference here?  Update the tests?  Or have the tests pass in a format string that doesn't include this option?

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb [hunk 2]
```diff
@@ -209,13 +209,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             If symbol.IsAnonymousType Then
                 AddAnonymousTypeName(symbol)
                 Return
-
             ElseIf symbol.IsTupleType Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb:212]: >ElseIf symbol.IsTupleType Then [](http://example.com/codeflow?start=12&length=30)

Consider restoring the empty line above

**Comment by AlekseyTs** [comment]
>  a bunch of tests change here because they were passing in a common error format with the "ExpandValueTuple" flag which VB previously didn't support. Now that it does, the tests all change in their error output (in generated IL comments). What's the preference here? Update the tests? Or have the tests pass in a format string that doesn't include this option?

I think IL verification base-lines should be updated. Are there other affected tests?

**Comment by AlekseyTs** [comment]
Done with review pass (commit 2)

**Comment by CyrusNajmabadi** [comment]
> I think IL verification base-lines should be updated. Are there other affected tests?

I don't believe so.  But i will call out if so after updating the baselines for the IL verification tests.  Thanks!

**Review by 333fred — APPROVED** [review]

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb [hunk 3]
```diff
@@ -870,7 +870,7 @@ End Module
   .maxstack  2
   IL_0000:  ldc.i4.2
   IL_0001:  ldc.i4.3
-  IL_0002:  call       "Function C.M1(Integer, Integer) As (Integer, Integer)"
+  IL_0002:  call       "Function C.M1(Integer, Integer) As ValueTuple(Of Integer, Integer)"
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb:873]: >ValueTuple [](http://example.com/codeflow?start=59&length=10)

I am curious why "System." is missing here. Is this expected? For example, it is present on the line below

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb [hunk 4]
```diff
@@ -906,22 +906,22 @@ hello
 {
   // Code size       67 (0x43)
   .maxstack  2
-  .locals init (System.ValueTuple(Of Integer, Integer, Integer, Integer, Integer, Integer, Integer, (Integer, Integer, Integer, Integer, String, Integer, Integer, Integer, Integer, String, Integer)) V_0) //t
+  .locals init (System.ValueTuple(Of Integer, Integer, Integer, Integer, Integer, Integer, Integer, ValueTuple(Of Integer, Integer, Integer, Integer, String, Integer, Integer, ValueTuple(Of Integer, Integer, String, Integer))) V_0) //t
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb:909]: >ValueTuple [](http://example.com/codeflow?start=100&length=10)

Here as well.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb:873]: good catch.  logic seems wrong vs c#.  i didn't grab all the things it does.

Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: Fixed.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: Fixed.

**Review by 333fred — APPROVED** [review]

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 7)

**Pull request merged**